DQM/SiPixelMonitorDigi/src/SiPixelDigiSource.cc

```diff
@@ -101,35 +103,94 @@ SiPixelDigiSource::~SiPixelDigiSource() {
   LogInfo("PixelDQM") << "SiPixelDigiSource::~SiPixelDigiSource: Destructor" << endl;
 }
 
-std::shared_ptr<bool> SiPixelDigiSource::globalBeginLuminosityBlock(const edm::LuminosityBlock& lumi,
-                                                                    const edm::EventSetup& iSetup) const {
+std::shared_ptr<SiPixelDigiCounter> SiPixelDigiSource::globalBeginLuminosityBlock(const edm::LuminosityBlock& lumi,
+                                                                                  const edm::EventSetup& iSetup) const {
   unsigned int currentLS = lumi.id().luminosityBlock();
   bool resetCounters = (currentLS % 10 == 0) ? true : false;
-  return std::make_shared<bool>(resetCounters);
+  auto lsc = std::make_shared<SiPixelDigiCounter>();
+
+  if (modOn && resetCounters && averageDigiOccupancy) {
```

Comment:
When can the `averageDigiOccupancy` part evaluate to `false`?

(I didn't see how it could be, but if I missed something, the `SiPixelDigiCounter` would likely have to split into two classes, one for counters being reset when `averageDigiOccupancy == true` and another being reset independently of `averageDigiOccupancy`)